### DISCUSSION TITLE
How to get blog excerpts to work

### DISCUSSION THREAD

**Discussion by Daeraxa** [vuepress-theme-hope/vuepress-theme-hope#2396] [Q&A] [answered]
Hi, I'm trying to set up the blog plugin so we can have a blog along with the rest of the documentation. I have *some* of it working but I just don't understand how to get the excerpt to work.

I think I have configured the `autoexcerpt` as per https://vuepress-theme-hope.github.io/v2/guide/blog/article.html#excerpt

I have also tried to follow the examples for the plugin options but the examples I find quite confusing as they all appear to be doing similar things but very differently.

For example:
https://vuepress-theme-hope.github.io/v2/blog/guide.html#gathering-info shows `categories` and `tags` (which I could not get to work) but the demo (https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/main/demo/blog2/src/.vuepress/config.ts) shows `category` and `tag`.

I currently have this in my `config.js`

```js
  theme: hopeTheme({
    ...
    plugins: {
    ...
      blog: {
        filter: ({ filePathRelative }) =>
          filePathRelative ? filePathRelative.startsWith("docs/blog/") : false,
        autoExcerpt: true,
        getInfo: ({ excerpt, frontmatter, git = {} }) => ({
          author: frontmatter.author || "",
          date: frontmatter.date || git.createdTime || null,
          category: frontmatter.category || [],
          tag: frontmatter.tag || [],
          excerpt: page.excerpt
        }),
      }
    },
```

The rest seems to work but I cannot get the auto excerpt to show. Nor can I seem to get it to work by adding an html comment into the doc nor into the frontmatter as per https://vuepress-theme-hope.github.io/v2/guide/blog/article.html#excerpt:

> If you want to add a excerpt for article, you can use  comment to mark it. All contents before the comment will be regarded as excerpt. Also, you can specific it with excerpt in frontmatter with a HTML string.

I've also tried commenting out `excerpt: page.excerpt` as the document implies that this is enabled by default but again, I cannot get anything to display.

What have I done wrong here and what do I need to do to display the automatic excerpt (or any excerpt)?

**Screenshot**
![image](https://user-images.githubusercontent.com/58074586/200982085-39529320-3f57-45c1-b051-76a71bc039e6.png)

**Test note content**
```md
---
title: Testing note 1
author: Daeraxa
date: 2022-11-10
category:
  - testcat
  - testcat2
tag:
  - testtag
  - testtag2
---
Some words
```

**Comment by Mister-Hope** [comment, marked as answer]
Some improvements and fixes are made in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/11e5a44e2697c01fc88379787036cd612920b2d8 and https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/012b9940acc8686bc7206822f6e433a756c28adb

**Reply by Daeraxa** [reply]
Thank you very much. How does the `autoExcerpt` work though? Is there something else that needs to be done to get it to display?
